Tests fail on Python 3.14

### Checklist

- [x] I'm reporting a bug unrelated to a specific site
- [x] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [x] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766), [the FAQ](https://github.com/yt-dlp/yt-dlp/wiki/FAQ), and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=is%3Aissue%20-label%3Aspam%20%20) for similar issues **including closed ones**. DO NOT post duplicates

### Provide a description that is worded well enough to be understood

We package yt-dlp in Fedora Linux and our development branch has been updated to Python 3.14. yt-dlp currently has one failing test on Python 3.14.

Downstream bug: https://bugzilla.redhat.com/show_bug.cgi?id=2345522

### Provide verbose output that clearly demonstrates the problem

- [x] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [x] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [x] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
=================================== FAILURES ===================================
_____________________ TestTraversal.test_traversal_morsel ______________________

self = <test.test_traversal.TestTraversal object at 0x7f56808cd0a0>

    def test_traversal_morsel(self):
        values = {
            'expires': 'a',
            'path': 'b',
            'comment': 'c',
            'domain': 'd',
            'max-age': 'e',
            'secure': 'f',
            'httponly': 'g',
            'version': 'h',
            'samesite': 'i',
        }
        morsel = http.cookies.Morsel()
        morsel.set('item_key', 'item_value', 'coded_value')
        morsel.update(values)
        values['key'] = 'item_key'
        values['value'] = 'item_value'
    
        for key, value in values.items():
            assert traverse_obj(morsel, key) == value, \
                'Morsel should provide access to all values'
>       assert traverse_obj(morsel, ...) == list(values.values()), \
            '`...` should yield all values'
E       AssertionError: `...` should yield all values
E       assert ['a', 'b', 'c...'e', 'f', ...] == ['a', 'b', 'c...'e', 'f', ...]
E         
E         At index 9 diff: '' != 'item_key'
E         Left contains one more item: 'item_value'
E         
E         Full diff:
E           [
E               'a',...
E         
E         ...Full output truncated (12 lines hidden), use '-vv' to show

test/test_traversal.py:439: AssertionError
```

--- a/test/test_traversal.py
+++ b/test/test_traversal.py
@@ -416,18 +416,8 @@ class TestTraversal:
             '`any` should allow further branching'
 
     def test_traversal_morsel(self):
-        values = {
-            'expires': 'a',
-            'path': 'b',
-            'comment': 'c',
-            'domain': 'd',
-            'max-age': 'e',
-            'secure': 'f',
-            'httponly': 'g',
-            'version': 'h',
-            'samesite': 'i',
-        }
         morsel = http.cookies.Morsel()
+        values = dict(zip(morsel, 'abcdefghijklmnop'))
         morsel.set('item_key', 'item_value', 'coded_value')
         morsel.update(values)
         values['key'] = 'item_key'

